### PR TITLE
Add trajectory reporter to openmm workflow

### PR DESCRIPTION
**Pull request by orionarcher**
## Summary

Include a summary of major changes in bullet points:

* Allows the OpenMM trajectories to report pymatgen trajectories

## Checklist

Work-in-progress pull requests are encouraged, but please put [WIP] in the pull request
title.

Before a pull request can be merged, the following items must be checked:

* [x] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
  The easiest way to handle this is to run the following in the **correct sequence** on
  your local machine. Start with running [`ruff`](https://docs.astral.sh/ruff) and `ruff format` on your new code. This will
  automatically reformat your code to PEP8 conventions and fix many linting issues.
* [x] Doc strings have been added in the [Numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
  Run [ruff](https://beta.ruff.rs/docs/rules/#pydocstyle-d) on your code.
* [x] Type annotations are **highly** encouraged. Run [mypy](http://mypy-lang.org) to
  type check your code.
* [x] Tests have been added for any new functionality or bug fixes.
* [x] All linting and tests pass.

Note that the CI system will run all the above checks. But it will be much more
efficient if you already fix most errors prior to submitting the PR. It is highly
recommended that you use the pre-commit hook provided in the repository. Simply run
`pre-commit install` and a check will be run prior to allowing commits.

### CODE REVIEW

**Comment by orionarcher** [comment]
Would you mind taking a look @janosh?

File: src/atomate2/openmm/utils.py [hunk 1]
```diff
@@ -171,3 +172,148 @@ def openff_to_openmm_interchange(
...
+        )
+        box_vectors = state.getPeriodicBoxVectors(asNumpy=True).value_in_unit(
+            omm_unit.angstrom
+        )
+
+        ev_to_kj_per_mol = 96.485
+
+        # Get energies in eV
+        kinetic_energy = (
+            state.getKineticEnergy()._value / ev_to_kj_per_mol  # noqa: SLF001
```
Comment (janosh) [inline src/atomate2/openmm/utils.py:269]: is there no public openMM API to get these values? seems fragile to rely on `_value` here which i'm guessing has no backwards compat gurantees?

File: src/atomate2/openmm/utils.py [hunk 2]
```diff
@@ -171,3 +172,148 @@ def openff_to_openmm_interchange(
...
+            {
+                "kinetic_energy": kinetic_energy,
+                "potential_energy": potential_energy,
+                "total_energy": kinetic_energy + potential_energy,
+            }
+        )
+
+        self._nextModel += 1
+
+    def __del__(self) -> None:
```
[the remark this answers is not in the document]
Comment (orionarcher) [reply src/atomate2/openmm/utils.py]: Good point. I was mirroring an external API and being a bit. I replaced it with a save method to be more robust. 

The core problem is that the `Reporters` usually append to a file at each step but trajectory doesn't easily support that.

File: src/atomate2/openmm/utils.py (hunk 1 above)
Comment (orionarcher) [reply src/atomate2/openmm/utils.py]: Good catch. I was a bit more clever and fixed it.

**Comment by orionarcher** [comment]
There are other `Reporters` that all use the camelCase args. I think OpenMM call's some of the methods internally too, so it's not possible to change them. I'd advocate for sticking with that, even though I agree it's a bit heretical.

**Comment by janosh** [comment]
looks good! thanks @orionarcher  👍

**Comment by orionarcher** [comment]
This will need one more loop back @janosh

**Pull request merged**